src/multilayerqg.jl

```diff
@@ -297,12 +341,17 @@ function Params(nlayers, g, f₀, β, ρ, H, U, eta, μ, ν, nν, grid; calcFq=n
     end
     CUDA.@allowscalar @views Qy[:, :, nlayers] = @. Qy[:, :, nlayers] - Fm[nlayers-1] * (U[:, :, nlayers-1] - U[:, :, nlayers])
 
-    return Params(nlayers, T(g), T(f₀), T(β), A(ρ), A(H), U, eta, T(μ), T(ν), nν, calcFq, A(g′), Qx, Qy, S, S⁻¹, rfftplanlayered)
+    if nlayers==2
+      return TwoLayerParams(T(g), T(f₀), T(β), A(ρ), (T(H[1]), T(H[2])), U, eta, T(μ), T(ν), nν, calcFq, T(g′[1]), Qx, Qy, rfftplanlayered)
```

Comment:
Just for my own understanding - Why would this make it much faster? It seems we are still calculating S and S-1 in lines 330-334 when nlayers==2, even though they are not needed?

Reply:
@szy21, this part is not what makes the difference. Either way computing or not the S and S^{-1} matrices only happens once so that was not what was causing the slowdown.

But by defining a new AbstractParameter type for 2-layers then we can add a new method to `streamfunctionfrompv!` that works specifically for 2-layers.

See, e.g., this general one
https://github.com/FourierFlows/GeophysicalFlows.jl/blob/4c310b84072207d0b73d2091e40f01abfab8da98/src/multilayerqg.jl#L560-L572

versus this 2-layer specific one that has the inversion hard-coded in.
https://github.com/FourierFlows/GeophysicalFlows.jl/blob/4c310b84072207d0b73d2091e40f01abfab8da98/src/multilayerqg.jl#L586-L618